src/maud/sampling.py

```diff
@@ -74,6 +75,53 @@ def get_full_stoichiometry(
     return S_enz, S_enz_to_flux_map, S_complete, S_drain
 
 
+def validate_specified_fluxes(
+    S_complete,
+    rxn_measurements,
+    experiment_codes: Dict[str, int],
+    reaction_codes: Dict[str, int],
+    drain_codes: Dict[str, int],
+    balanced_mic_codes: Dict[str, int],
+):
+    """Validate measured fluxes.
+
+    :param S_complete: The stoichiometrix matrix including reactions and drains
+    :param rxn_measurements: A dataframe containing all flux measurements
+    :param experiment_codes: the codified experiment codes
+    :param reaction_codes: the codified reaction codes
+    :param drain_codes: the codified drain codes
+    :param balanced_mic_codes: the codified balanced_mic codes
+    """
+    complete_reactions = list(reaction_codes.keys()) + list(drain_codes.keys())
+    for exp in experiment_codes:
+        meas_rxns = rxn_measurements[rxn_measurements["experiment_id"] == exp][
+            "target_id"
+        ]
+        measured_rxn_list = list(meas_rxns)
+        flux_paths = null(S_complete[balanced_mic_codes.keys()].T.values)
+        n_rxns, n_dof = np.shape(flux_paths)
+        rref_flux_paths = np.matrix(rref(flux_paths.T))
+        rref_flux_paths[np.abs(rref_flux_paths) < 1e-10] = 0
+        flux_paths_df = pd.DataFrame(rref_flux_paths, columns=complete_reactions)
+        for _, flux_path in flux_paths_df.iterrows():
+            if any(flux_path[measured_rxn_list]) != 0:
+                pass
+            else:
+                possible_measurements = []
+                for rxn, st in flux_path.items():
+                    if st != 0:
+                        possible_measurements.append(rxn)
+                print("Your system appears to be underdetermined in experiment:")
+                print(f"{exp}")
+                print("Please define a reaction from the following list:")
+                print(f"{possible_measurements}")
+
```

Comment:
define a reaction from the following list >> specify a measurement of one of the following reactions

I think this is a good time to use the [warnings](https://docs.python.org/3/library/warnings.html#warnings.warn) module:

```
possible_measurements = [rxn for rxn, st in flux_path.items() if st != 0]
msg = (
    f"Your system appears to be underdetermined in experiment:\n{exp}\n" +
    "Please define a reaction from the following list:\n"+
    f"{'\n'.join(possible_measurements)}"
)
warnings.warn(msg)
```
(this will also work for the check below)